.circleci/config.yml

```diff
@@ -0,0 +1,45 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/python:3.6
+    steps:
+      - checkout
+
+      - restore_cache:
+          key: dependency-cache-{{ checksum "requirements.txt" }}
```

Comment:
This means we lose the whole virtualenv when a dependency is added or changed.  In my projects I have keys like `project-v1` and I bump the number when I change cache paths, base image or CI tools.

Reply:
Hey, 

> This means we lose the whole virtualenv when a dependency is added or changed.

I am not quite sure I understand what you mean. If you add a dependency, `pip` is still being ran so it will always install all dependencies. If it is found in the cache it will be "instant", if not it will be downloaded. 

>In my projects I have keys like project-v1 and I bump the number when I change cache paths

I am not quite sure I understand the difference between this and your example. The key can be any arbitrary thing. I just named it `dependency-cache`. In addition you can have [multiple cache keys](https://circleci.com/docs/2.0/caching/#using-keys-and-templates) which increases the chance of a cache being hit.

Reply:
I mean the previous cache won’t be restored, so pip will have to install all dependencies fresh.  In contrast, a key without requirements file checksum will cause the cache to be restored, and then pip will install only the one new dependency.